engine/schema/src/com/cloud/upgrade/dao/Upgrade481to490.java

```diff
@@ -53,6 +62,139 @@ public boolean supportsRollingUpgrade() {
 
     @Override
     public void performDataMigration(Connection conn) {
+        setupRolesAndPermissionsForDynamicRBAC(conn);
+    }
+
+    private void createDefaultRole(final Connection conn, final Long id, final String name, final RoleType roleType) {
+        final String insertSql = String.format("INSERT INTO `cloud`.`roles` (`id`, `uuid`, `name`, `role_type`, `description`) values (%d, UUID(), '%s', '%s', 'Default %s role');",
+                id, name, roleType.name(), roleType.name().toLowerCase());
+        try ( PreparedStatement updatePstmt = conn.prepareStatement(insertSql) ) {
+            updatePstmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new CloudRuntimeException("Unable to create default role with id: " + id + " name: " + name, e);
+        }
+    }
+
+    private void createRoleMapping(final Connection conn, final Long roleId, final String apiName) {
+        final String insertSql = String.format("INSERT INTO `cloud`.`role_permissions` (`uuid`, `role_id`, `rule`, `permission`) values (UUID(), %d, '%s', 'ALLOW') ON DUPLICATE KEY UPDATE rule=rule;",
+                roleId, apiName);
+        try ( PreparedStatement updatePstmt = conn.prepareStatement(insertSql)) {
+            updatePstmt.executeUpdate();
+        } catch (SQLException ignored) {
+            s_logger.debug("Unable to insert mapping for role id:" + roleId + " apiName: " + apiName);
+        }
+    }
+
+    private void addRoleColumnAndMigrateAccountTable(final Connection conn, final RoleType[] roleTypes) {
+        final String alterTableSql = "ALTER TABLE `cloud`.`account` ADD COLUMN `role_id` bigint(20) unsigned COMMENT 'role id for this account' AFTER `type`, " +
+                "ADD KEY `fk_account__role_id` (`role_id`), " +
+                "ADD CONSTRAINT `fk_account__role_id` FOREIGN KEY (`role_id`) REFERENCES `roles` (`id`);";
+        try (PreparedStatement pstmt = conn.prepareStatement(alterTableSql)) {
+            pstmt.executeUpdate();
+            s_logger.info("Altered cloud.account table and added column role_id");
+        } catch (SQLException e) {
+            if (e.getMessage().contains("role_id")) {
+                s_logger.warn("cloud.account table already has the role_id column, skipping altering table and migration of accounts");
+                return;
+            } else {
+                throw new CloudRuntimeException("Unable to create column quota_calculated in table cloud_usage.cloud_usage", e);
+            }
+        }
+        migrateAccountsToDefaultRoles(conn, roleTypes);
+    }
+
+    private void migrateAccountsToDefaultRoles(final Connection conn, final RoleType[] roleTypes) {
+        try (PreparedStatement selectStatement = conn.prepareStatement("SELECT `id`, `type` FROM `cloud`.`account`;");
+             ResultSet selectResultSet = selectStatement.executeQuery()) {
```

Comment:
The ``selectResultSet`` is a resource that needs to be closed.  Please add it to enclosing try with resources block.